app/com/arpnetworking/metrics/portal/scheduling/JobRef.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import models.internal.Organization;
+import models.internal.scheduling.Job;
+import net.sf.oval.constraint.NotNull;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * A serializable reference to a {@link Job}.
+ *
+ * @param <T> The type of the result computed by the job.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class JobRef<T> implements Serializable, Supplier<Optional<Job<T>>> {
+    private final JobRepository<T> _repository;
```

Comment:
The `Repository` is not serializable, and therefore the the `JobRef` is not serializable either. I think you should should a type (e.g. Class<? extends JobRepository> or other repository type identifier) from which a recipient of `JobRef` can use Guice to obtain the correct repository reference.